src/types/any.rs

```diff
@@ -36,6 +37,12 @@ use std::os::raw::c_int;
 #[repr(transparent)]
 pub struct PyAny(UnsafeCell<ffi::PyObject>);
 
+// TODO: this will become the main PyAny type in PyO3 0.20, with normal ownership
+// rather than GIL-bound pool semantics.
+//
+// All other types to be migrated in individual PRs first.
+pub(crate) struct PyAnyOwned<'py>(NonNull<ffi::PyObject>, Python<'py>);
```

Comment:
Just out of curiosity, are there any validity issues which prevent this from just being a newtype wrapper around `&'py ffi::PyObject`?

Reply:
Ah, yes, pervasive interior mutability, so `&'py UnsafeCell<ffi::PyObject>` if anything...

Reply:
And indeed, this is not much more useful than `NonNull` and having the token "present" is nice in itself. Sorry for the noise.